portal/models.py

```diff
@@ -15,29 +18,49 @@ def __str__(self):
 
 
 class Student(models.Model):
-    username = models.CharField(max_length=7,unique=True)
+    GENDER_CHOICES = (
+        ('M', 'Male'),
+        ('F', 'Female'),
+    )
+
+    COURSE_CHOICES = (
+        ('C', 'Computing'),
+        ('J', 'JMC')
+    )
+
+    name = models.CharField(max_length=100)
+    username = models.CharField(max_length=7, unique=True)
+    gender = models.CharField(max_length=1, choices=GENDER_CHOICES)
+    course = models.CharField(max_length=1, choices=COURSE_CHOICES)
     magic_id = models.CharField(max_length=8)
     child = models.BooleanField()
     party = models.BooleanField(default=False)
     hobbies = models.ManyToManyField(to=Hobby, blank=True)
     partner = models.ForeignKey('Student', null=True, blank=True)
     family = models.ForeignKey('Family', null=True, blank=True)
     confirmed = models.BooleanField(default=False)
+    activated = models.BooleanField(default=False)
 
     def __str__(self):
-        return self.username
+        return self.name
 
     def save(self, *args, **kwargs):
         if not self.magic_id or len(self.magic_id) == 0:
             self.magic_id = get_random_id()
         super(Student, self).save(*args, **kwargs)
 
     def get_new_student_popup(self):
-        message = "Email has been sent to %s@ic.ac.uk. Go activate your account." % self.username
-        return {'message': message, 'state': 'success'}
+        popup = "Email has been sent to %s@ic.ac.uk. Go activate your account." % self.username
+        message = "To view the Mums and Dads portal go to %s/preferences/%s/\n\nKeep this URL safe, anyone with it " \
```

Comment:
Even in the previous review, I found “Go activate your account” a little passive aggressive, maybe “Please check your email to activate your account” would be better, imho.